sdk/core/azure-amqp/_connection.py

```diff
@@ -205,17 +207,19 @@ def _set_state(self, new_state: ConnectionState) -> None:
         :param ~pyamqp.constants.ConnectionState new_state: The new state of the connection.
         """
         if new_state is None:
-            return
-        previous_state = self.state
-        self.state = new_state
-        _LOGGER.info(
-            "Connection state changed: %r -> %r",
-            previous_state,
-            new_state,
-            extra=self._network_trace_params
-        )
-        for session in self._outgoing_endpoints.values():
-            session._on_connection_state_change()  # pylint:disable=protected-access
+                return
+        
+        with self._connection_lock:
```

Comment:
`_set_state` is called a lot, but it does modify self.state which is used a lot too. Could be a lock of acquire/release here but it might be needed